### ISSUE TITLE
Sanitizer-sparc-Test doesn't link on Solaris

### ISSUE THREAD

**Issue by rorth** [llvm/llvm-project#53704]
The `Sanitizer-sparc-Test` doesn't link on 32-bit Solaris/sparcv9 on the `release/14.x` branch:
```
Undefined                       first referenced
 symbol                             in file
__atomic_compare_exchange_8         SANITIZER_TEST_OBJECTS.sanitizer_addrhashmap_test.cpp.sparc.o
__atomic_load_8                     SANITIZER_TEST_OBJECTS.sanitizer_addrhashmap_test.cpp.sparc.o
ld: fatal: symbol referencing errors
```
This is fixed on `main` now by [D118021](https://reviews.llvm.org/D118021); this patch needs to go onto the branch, too.

/cherry-pick a6afa9e6b0d9a08ecc878e62343a2af80ff30435

**Comment by tstellar** [comment]
@MaskRay What do you think about backporting this? https://github.com/llvm/llvm-project/commit/a6afa9e6b0d9a08ecc878e62343a2af80ff30435

**Comment by MaskRay** [comment]
LGTM

**Comment by tstellar** [comment]
Merged: 2fe5bf57172c
